lib/utils/background_push.dart

```diff
@@ -75,25 +75,28 @@ class BackgroundPush {
     onRoomSync ??= client.onSync.stream
         .where((s) => s.hasRoomUpdate)
         .listen((s) => _onClearingPush(getFromServer: false));
-    fcmSharedIsolate?.setListeners(onMessage: (message) {
-      Logs().d('BackgroundPush::onMessage(): $message');
-      final notification = _parseMessagePayload(message);
-      pushHelper(
-        notification,
-        client: client,
-        l10n: l10n,
-        activeRoomId: router?.currentState?.pathParameters['roomid'],
-        onSelectNotification: goToRoom,
-      );
-      Logs().d('BackgroundPush::onMessage(): finished pushHelper');
-    });
     if (Platform.isAndroid) {
       UnifiedPush.initialize(
         onNewEndpoint: _newUpEndpoint,
         onRegistrationFailed: _upUnregistered,
         onUnregistered: _upUnregistered,
         onMessage: _onUpMessage,
       );
+      fcmSharedIsolate?.setListeners(onMessage: (message) {
+        onReceiveNotification(message);
+      });
+    } else if (Platform.isIOS) {
+      apnChannel.setMethodCallHandler((call) async {
+        Logs().v('[Push] Received APN call: $call');
+        if (call.method == 'willPresent') {
+          onReceiveNotification(call.arguments);
+        } else if (call.method == 'didReceive') {
+          onReceiveNotification(call.arguments);
+          // Go to room because user selected the notification
+          goToRoom(call.arguments['room_id']);
+        }
```

Comment:
- if `willPresent ` => Why need handle this case ?
- if `didReceive ` => Show local notification => Click on noti => go to Room right?

IMO: I'm not sure, can you explain this case when we handle it at `dart` layer?

Reply:
It changed, let me know if you still need explain